src/hooks/useLocalStorage.ts

```diff
@@ -1,32 +1,71 @@
 import { useState } from "react";
 
+type LocalStorageValue = number | string | boolean | {};
+export type LocalStorageInitCb<T extends LocalStorageValue> = (
+  initial: unknown
```

Comment:
```suggestion
  initial: T | null
```

Reply:
It can be anything - `string`, `number` or `object` so i put `unknown` here

Reply:
We should establish that the provided type is what's currently in localStorage (or null if it hasn't been defined). Instead if user chooses not to provide the type, we could then fallback to unknown:
```
export default function useLocalStorage<T = unknown>(
```